noise2same/network.py

```diff
@@ -490,36 +491,58 @@ def __init__(
             kernel_size=kernel_size,
             ffc=ffc,
             unit_type=unit_type,
-            global_ratio=global_ratio)
+            global_ratio=global_ratio,
+        )
 
     def forward(self, x: T) -> T:
         if self.unit_type == "dual_pass":
             x_l, x_g = x
-            x_l = self.downsampling_block_local(x_l)
-            x_g = self.downsampling_block_global(x_g)
+            x_l = self.twin_unit.sample_local(x_l)
+            x_g = self.twin_unit.sample_global(x_g)
             x = (x_l, x_g)
         else:
             x = self.downsampling_block(x)
         x = self.resblock(x)
         return x
 
 
+class TwinSample(nn.Module):
```

Comment:
you can also add `self.forward()` method to call both `self.sample_local()` and `self.sample_global()` and return a tuple `(x_l, x_g)`. Then the signature will be the same as for the `downsampling_block` in line 504 above